### PR TITLE
MM-23782 - fix site statistics ordering

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
We attempt to create all 3 charts with the same "label" data. (ie. Dates). However, when missing dates are added, they are added to the end of the list, not in date order. 

This PR sorts the dates prior to filling in the chart data with the missing dates. 

The "dates" are returned from the server as string labels. They are sortable by strings as they always return in 'yyyy-mm-dd' format. (ie. 2020-06-15)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23782

#### Screenshots
Screen Shot 2020-06-23 at 4 41 40 PM

### CODE REVIEW

File: components/analytics/format.tsx [hunk 1]
```diff
@@ -49,43 +49,22 @@ export function formatPostsPerDayData(data: any) {
...
-    chartDatas.forEach((chartData) => {
-        if (chartData.labels.length > 0) { // don't add to empty graphs
-            labels.forEach((label: string) => {
-                if (chartData.labels.indexOf(label) === -1) {
-                    chartData.labels.push(label);
-                    chartData.datasets[0].data.push(0);
-                }
-            });
+    datas.forEach((data: any) => {
+        if (data && data.length > 0) {
```
Comment (hahmadia) [inline components/analytics/format.tsx:60]: Could this become
```suggestion
        if (data?.length) {
```
?

File: components/analytics/format.tsx [hunk 2]
```diff
@@ -99,14 +78,19 @@ export function formatUsersWithPostsPerDayData(data: any) {
...
 
-    for (const index in data) {
-        if (data[index]) {
-            const row = data[index];
-            chartData.labels.push(row.name);
-            chartData.datasets[0].data.push(row.value);
-        }
-    }
+function fillChartData(chartData: any, labels: any, data: any) {
+    if (data && data.length > 0) {
```
Comment (hahmadia) [inline components/analytics/format.tsx:85]:
```suggestion
    if (data?.length) {
``` 
Does this work ?

File: components/analytics/format.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ export function formatPostDoughtnutData(filePosts: any, hashtagPosts: any, total
     return postTypeData;
 }
 
-export function formatPostsPerDayData(data: any) {
+export function formatPostsPerDayData(labels: string[], data: any) {
     const chartData = {
         labels: [] as any,
```
Comment (hahmadia) [inline components/analytics/format.tsx:41]:
```suggestion
        labels: [] as string[],
```

File: components/analytics/format.tsx [hunk 4]
```diff
@@ -49,43 +49,22 @@ export function formatPostsPerDayData(data: any) {
...
+            data.forEach((e: any) => labels.add(e.name));
         }
     });
+    return Array.from(labels).sort();
 }
 
-export function formatUsersWithPostsPerDayData(data: any) {
+export function formatUsersWithPostsPerDayData(labels: string[], data: any) {
     const chartData = {
         labels: [] as any,
```
Comment (hahmadia) [inline components/analytics/format.tsx:69]:
```suggestion
        labels: [] as string[],
```

File: components/analytics/format.tsx [hunk 5]
```diff
@@ -99,14 +78,19 @@ export function formatUsersWithPostsPerDayData(data: any) {
...
-            chartData.labels.push(row.name);
-            chartData.datasets[0].data.push(row.value);
-        }
-    }
+function fillChartData(chartData: any, labels: any, data: any) {
+    if (data && data.length > 0) {
+        chartData.labels = labels;
 
+        //labels are in order, add in label order...
+        chartData.labels.forEach((label: any) => {
```
Comment (hahmadia) [inline components/analytics/format.tsx:89]:
```suggestion
        chartData.labels.forEach((label: string) => {
```

File: components/analytics/format.tsx (hunk 2 above)
Comment (sbishel) [reply components/analytics/format.tsx]: added separately, because I ran into an issue before an wanted to test. But these suggestions all work...added.

**Pull request merged**